tests/common/plugins/sanity_check/checks.py

```diff
@@ -311,184 +309,198 @@ def _check_monit_services_status(check_result, monit_services_status):
-    # Send dummy ARP packets from PTF to ToR. Ensure that ARP is learned on both ToRs
-    failed = False
-    reason = ''
-    host = None
-    upper_tor_host.shell("ip neigh flush all")
-    lower_tor_host.shell("ip neigh flush all")
+def _check_tor_flows(active_flows, mux_intf, bridge):
+    '''
+    Checks that the flows originationg from the active ToR are correct
 
-    testutils.send_packet(ptfadapter, ptf_port_index, ptf_arp_pkt)
+    1. Checks that there is exactly one flow (for the server/NIC)
+    2. Checks that the action is output
+    3. Checks that the out port is the server/NIC interface
+    '''
+    failed = False
+    failed_reason = ''
 
-    upper_tor_arp_table = upper_tor_host.switch_arptable()['ansible_facts']['arptable']['v4']
-    lower_tor_arp_table = lower_tor_host.switch_arptable()['ansible_facts']['arptable']['v4']
-    if ptf_arp_tgt_ip not in upper_tor_arp_table:
+    # Checking active ToR flows
+    if len(active_flows) != 1:
+        failed = True
+        failed_reason = 'Incorrect number of active ToR flows for {}, got {} but expected 1' \
+                        .format(bridge, len(active_flows))
+        return failed, failed_reason
+    
+    if active_flows[0]['action'] != 'output':
+        failed = True
+        failed_reason = 'Incorrect active ToR action for {}, got {} but expected output' \
+                        .format(bridge, active_flows[0]['action'])
+        return failed, failed_reason
+    
+    if active_flows[0]['out_port'] != mux_intf:
         failed = True
-        reason = 'Packet from PTF not received on upper ToR {}'.format(upper_tor_host)
-        host = upper_tor_host.hostname
-        return failed, reason, host
+        failed_reason = 'Incorrect active ToR flow output interface for {}, got {} but expected {}' \
+                        .format(bridge, active_flows[0]['out_port'], mux_intf)
+        return failed, failed_reason
+    
+    return failed, failed_reason
 
-    if ptf_arp_tgt_ip not in lower_tor_arp_table:
+
+def _check_single_intf_status(intf_status, expected_side):
+    """
+    Checks the mux simulator status for a single ToR/server connection
+    """
+    failed = False
+    failed_reason = ''
+
+    bridge = intf_status['bridge']
+
+    # Check the total number of flows is 2, one for 
+    # server to both ToRs and one for active ToR to server
+    if len(intf_status['flows']) != 2:
         failed = True
-        reason = 'Packet from PTF not received on lower ToR {}'.format(lower_tor_host)
-        host = lower_tor_host.hostname
+        failed_reason = 'Incorrect number of flows for {}, got {} but expected 2' \
+                        .format(bridge, len(intf_status['flows']))
+        return failed, failed_reason
+
+    # Gather the flow information
+    active_intf, mux_intf = None, None
+    active_flows, mux_flows = None, None
+
+    for input_intf, actions in intf_status['flows'].items():
+        if 'muxy' in input_intf:
```

Comment:
With the fix of https://github.com/Azure/sonic-mgmt/pull/3359, PTF interface name is not guaranteed to be starting with "muxy". Suggest to use "mu".